neurodsp/sim/cycles.py

```diff
@@ -204,6 +204,109 @@ def sim_gaussian_cycle(n_seconds, fs, std):
     return cycle
 
 
+def sim_gaussian_asym_cycle(n_seconds, fs, beta, scale=2, shift=1):
+    """Simulate an asymmetrical guassian cycle.
+
+    Parameters
+    ----------
+    n_seconds : float
+        Length of cycle window in seconds.
+    fs : float
+        Sampling frequency of the cycle simulation.
+    beta : float
+        Exponent controlling the amplitude asymmetry of peaks relative to the troughs.
+
+        - `beta=0` : zeros
+        - `beta=.5`: wide peaks, narrow troughs
+        - `beta=1.`: symetrical peaks and troughs (sine wave)
+        - `beta=5.`: wide troughs, narrow peaks
+
+    scale : float, optional, default: 2
+        Rescales the amplitude of the signal.
+    shift : float, optional, default: 1
+        Translate the signal along the y-axis.
+
+    Returns
+    -------
+    cycle : 1d array
+        Simulated asymmetrical gaussian cycle.
+
+    Notes
+    -----
+    If shift or scale keyword arguments are given, default signal normalization will be bypassed.
+
+    References
+    ----------
+    Lozano-Soldevilla, D., Huurne, N. T., &amp; Oostenveld, R. (2016). Neuronal Oscillations with
+    Non-sinusoidal Morphology Produce Spurious Phase-to-Amplitude Coupling and Directionality.
+    Frontiers in Computational Neuroscience, 10. doi:10.3389/fncom.2016.00087
+
+    Examples
+    --------
+    Simulate a cycle of an asymmetrical gaussian wave:
+
+    >>> cycle = sim_gaussian_asym_cycle(n_seconds=1, fs=500, beta=3)
+    """
+
+    check_param(beta, 'beta', [0., np.inf])
+
+    times = create_cycle_time(n_seconds, fs)
+
+    cycle = ((-np.cos(times) + shift) / scale)**beta
+
+    return cycle
+
+
+def sim_harmonic_asym_cycle(n_seconds, fs, phi, n_harmonics):
+    """Simulate an asymmetrical cycle as a sum of harmonics.
+
+    Parameters
+    ----------
+    n_seconds : float
+        Length of cycle window in seconds.
+    fs : float
+        Sampling frequency of the cycle simulation.
+    phi : float
+        Phase at each harmonic.
+    n_harmonics : int
+        Number of harmonics to sum across.
+
+    Returns
+    -------
+    cycle : 1d array
+        Simulated asymmetrical harmonic cycle.
+
+    Notes
+    -----
+    .. math::
+
+      cycle = \sum_{j=1}^{j} \dfrac{1}{j^2} \cdot cos(j2\pi ft)+(j-1)*\phi
+
+    References
+    ----------
+    Lozano-Soldevilla, D., Huurne, N. T., &amp; Oostenveld, R. (2016). Neuronal Oscillations with
+    Non-sinusoidal Morphology Produce Spurious Phase-to-Amplitude Coupling and Directionality.
+    Frontiers in Computational Neuroscience, 10. doi:10.3389/fncom.2016.00087
+
+    Examples
+    --------
+    Simulate an asymmetrical cycle as the sum of harmonics:
+
+    >>> cycle = sim_gaussian_asym_cycle(n_seconds=1, fs=500, beta=3)
+    """
+    times = create_cycle_time(n_seconds, fs)
+    cycs = np.zeros((n_harmonics+1, len(times)))
+
+    harmonics = np.array([harmonic for harmonic in range(1, n_harmonics + 2)])
```

Comment:
I think this can be simplified to `np.array([range(1, n_harmonics + 2)])`?

Reply:
Good idea! I didn't realize ranges could be passed like this. One note for when I come back to make this round of updates: the brackets will need to be removed so it's not a 2D array.